test_util/launch.py

```diff
@@ -1,19 +1,145 @@
-"""
+"""DC/OS Launch
 
 Usage:
-dcos-launch create [--wait] [--dump-info=cluster_info.json] <config=config.yaml>
-dcos-launch wait <path=cluster_info.json>
-dcos-launch describe <path=cluster_info.json>
-dcos-launch delete <path=cluster_info.json>
-"""
+  dcos-launch create [--info-path=<path>] [--config-path=<path>]
+  dcos-launch wait [--info-path=<path>]
+  dcos-launch (describe|delete) [--info-path=<path>]
 
+Commands:
+  create    reads the file given by --config-path, creates the cluster described therein
+            and finally dumps a JSON file to the path given in --info-path which can then
+            be used with the wait, describe, and delete calls
+  wait      Block until the cluster is up and running
+  describe  Return additional information about the composition of the cluster
+  delete    Destroying the provided cluster deployment
+
+Options:
+  --config-path=<path>  Path for config to create cluster from [default: config.yaml].
+  --info-path=<path>    JSON file output by create and consumed by wait, describe,
+                        and delete [default: cluster_info.json].
+"""
+import abc
+import json
+import os
+import sys
+import yaml
 from docopt import docopt
+from pkgpanda.util import load_json, load_string, write_json
+
+from test_util.aws import BotoWrapper, DcosCfSimple
+
+
+class AbstractLauncher(metaclass=abc.ABCMeta):
+    def __init__(self):
```

Comment:
I don't think there is value in forcing Launchers to have a custom `__init__` (as this does). If we had something like a `dcos-docker` or `dcos-vagrant` launcher, those would have less or very different things (if anything in the init)